source/common/router/router.cc

```diff
@@ -819,6 +874,15 @@ Http::FilterTrailersStatus Filter::decodeTrailers(Http::RequestTrailerMap& trail
   if (!upstream_requests_.empty()) {
     upstream_requests_.front()->acceptTrailersFromRouter(trailers);
   }
+  if (streaming_shadows_) {
+    for (auto* shadow_stream : shadow_streams_) {
```

Comment:
Are you able to just iterate over shadow_streams_ without the first check? Wouldn't it be empty if disabled?

Reply:
Removed.